main.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/shopspring/decimal"
+)
+
+type membership struct {
+	references map[string][]string
+	members    map[string]member
+	newMembers []member
+}
+
+type transactions struct {
+	ignored   []bankTxn
+	incorrect []bankTxn
+	candidate []bankTxn
+	unmatched []bankTxn
+}
+
+type members struct {
+	paying       []member
+	unmatchedIDs []string
+}
+
+type activeMembers struct {
+	txns    transactions
+	members members
+}
+
+func newMembership(referencePath, membersPath, newMembersPath string) (*membership, error) {
+	m := membership{}
+	references, err := loadMemberReferencesFromCsv(referencePath)
+	if err != nil {
+		return nil, err
+	}
+
+	m.references = references
+	members, err := loadMembershipDetailsFromCsv(membersPath)
+	if err != nil {
+		return nil, err
+	}
+
+	m.members = members
+	newMembers, err := loadNewMembersFromCsv(newMembersPath)
+	if err != nil {
+		panic(err)
+	}
+
+	m.newMembers = newMembers
+
+	return &m, nil
+}
+
+func (m *membership) loadAndFilterTxns(txnsPath string, correctMembershipAmounts, otherMembershipAmounts []decimal.Decimal) (am *activeMembers, err error) {
+	txns, err := loadTxnsFromCsv(txnsPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var transactions = transactions{}
+	membershipAmounts := append(correctMembershipAmounts, otherMembershipAmounts...)
+	transactions.candidate, transactions.ignored = filterInterestingTxns(txns, membershipAmounts)
+	_, transactions.incorrect = filterInterestingTxns(transactions.candidate, correctMembershipAmounts)
+	var memberIds []string
+	memberIds, transactions.unmatched = identifyMembers(transactions.candidate, m.references)
+	var members = members{}
+	members.paying, members.unmatchedIDs = matchMembers(m.members, memberIds)
+
+	am = new(activeMembers)
+	am.txns = transactions
+	am.members = members
+
+	return am, err
+}
+
+func main() {
+	if len(os.Args) != 5 {
+		panic("Missing argument!")
+	}
+
+	correctMembershipAmounts := []decimal.Decimal{
+		decimal.New(185, -1),
+		decimal.New(30, 0),
+	}
+	membershipAmounts := []decimal.Decimal{
+		decimal.New(165, -1),
+		decimal.New(15, 0),
+		decimal.New(25, 0),
+	}
+	membershipAmounts = append(membershipAmounts, correctMembershipAmounts...)
+	ignoreTxnsPath := "ignored_txns.csv"
+	incorrectMembershipTxnsPath := "incorrect_membership_txns.csv"
+	unmatchedTxnsPath := "unmatched_txns.csv"
+	paidMembersPath := "paid_members.csv"
+	unmatchedMemberIDsPath := "unmatched_memberids.csv"
+	allMembersPath := "all_members.csv"
```

Comment:
These could be constants, declared above the main function.

```
const (
    DefaultignoreTxnsPath = "ignored_txns.csv"
    DefaultIncorrectMembershipTxnsPath = "incorrect_membership_txns.csv"
)
```

I've prefixed with "default" in case you ever intend overriding them at some point. If so, that prefix is commonly used.

Using constants has the added benefit that they get their own section whenever somebody looks up the source with GoDoc. Like this: https://golang.org/pkg/net/http/#pkg-constants

Reply:
Done.